utils/utf8.cc

```diff
@@ -56,22 +56,30 @@ namespace utils {
 
 namespace utf8 {
 
-// 3x faster than boost utf_to_utf
-static inline std::optional<size_t> validate_naive(const uint8_t *data, size_t len) {
-    size_t pos = 0;
-
-    while (len) {
-        size_t bytes;
-        const uint8_t byte1 = data[0];
+struct codepoint_status {
+    size_t bytes_validated;
+    bool error;
+    uint8_t more_bytes_needed;
+};
 
+static
+codepoint_status
+inline
+evaluate_codepoint(const uint8_t* data, size_t len) {
+    auto byte1 = data[0];
```

Comment:
const?

Reply:
Why? const it useful on interfaces, it tells the reader of the interface that the callee does not change the value.

How does it help here?

Reply:
when a variable is const a reader knows it will never change. When it's not const a reader has to read the whole code to know.
If something is const like byte1 in our case why not mark it as such?

Reply:
Ok, I'll add const. I think it's noise in small functions, but if it helps someone, it's worth it.